### PR TITLE
[DOCS] Refactor glossary

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:
* Edits each definition down to approximately 1-2 sentences.
* Adds a reference link to each definition.
* Removes the `type` definition.
* Adds new definitions for:
  * `merge`
  * `segment`
  * `token`
  * `tokenization`

### Preview
https://elasticsearch_70516.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/glossary.html

### CODE REVIEW

File: docs/reference/glossary.asciidoc [hunk 1]
```diff
@@ -1,670 +1,530 @@
...
-<<glossary-rollover,rollover>> to reduce the shards in the old index to a single segment.
 See the {ref}/indices-forcemerge.html[force merge API].
 // end::force-merge-def[]
 
-[[glossary-freeze]] freeze ::
+[[glossary-freeze]] freeze::
 // tag::freeze-def[]
 // tag::freeze-def-short[]
-Make an index read-only and minimize its memory footprint.
+Makes an index read-only and minimize its memory footprint.
```
Comment (lockewritesdocs) [inline docs/reference/glossary.asciidoc:161]:
```suggestion
Makes an index read-only and minimizes its memory footprint.
```

File: docs/reference/glossary.asciidoc [hunk 2]
```diff
@@ -1,670 +1,530 @@
...
 // end::frozen-tier-def[]
 
-[[glossary-hidden-index]] hidden index ::
+[[glossary-hidden-index]] hidden data stream or index::
 // tag::hidden-index-def[]
-An index that is excluded by default when you access indices using a wildcard expression.
-You can specify the `expand_wildcards` parameter to include hidden indices.
-Note that hidden indices _are_ included if the wildcard expression starts with a dot, for example `.watcher-history*`.
+<<glossary-data-stream,Data stream>> or <<glossary-index,index>> excluded from
+most <glossary-index-pattern,index patterns>> by default. See
```
Comment (lockewritesdocs) [inline docs/reference/glossary.asciidoc:192]:
```suggestion
most <<glossary-index-pattern,index patterns>> by default. See
```

File: docs/reference/glossary.asciidoc [hunk 3]
```diff
@@ -1,670 +1,530 @@
...
 // end::hot-phase-def[]
 
 [[glossary-hot-tier]] hot tier::
 // tag::hot-tier-def[]
-A <<glossary-data-tier, data tier>> that contains nodes that handle the indexing load
-for time series data such as logs or metrics and hold your most recent,
-most-frequently-accessed data.
+<<glossary-data-tier,Data tier>> that contains <<glossary-node,nodes>> that
+handle the <<glossary-index,indexing>> load for time series data such as logs or
+metrics and hold your most recent, most-frequently-accessed data. See
```
Comment (lockewritesdocs) [inline docs/reference/glossary.asciidoc:207]:
```suggestion
<<glossary-data-tier,Data tier>> that contains <<glossary-node,nodes>> that
handle the <<glossary-index,indexing>> load for time series data, such as logs or
metrics. This tier holds your most recent, most-frequently-accessed data. See
```

File: docs/reference/glossary.asciidoc [hunk 4]
```diff
@@ -1,670 +1,530 @@
...
-destination can be any pre-existing index, index alias, or
-{ref}/data-streams.html[data stream].
-
-You can reindex all documents from a source or select a subset of documents to
-copy. You can also reindex to a destination in a remote cluster.
-
-A reindex is often performed to update mappings, change static index settings,
-or upgrade {es} between incompatible versions.
+Copies documents from a source to a destination. The source and destination can
+be an <<glossary-data-stream,data stream>>, <<glossary-index,index>>, or
```
Comment (lockewritesdocs) [inline docs/reference/glossary.asciidoc:325]:
```suggestion
be a <<glossary-data-stream,data stream>>, <<glossary-index,index>>, or
```

File: docs/reference/glossary.asciidoc [hunk 5]
```diff
@@ -1,670 +1,530 @@
...
-<<glossary-term,terms>>, which is what is actually stored in the index.
-
-Text <<glossary-field,fields>> need to be analyzed at index time in order to
-be searchable as full text, and keywords in full text queries must be
-analyzed at search time to produce (and search for) the same terms
-that were generated at index time.
-
-See also <<glossary-term,term>> and <<glossary-analysis,analysis>>.
+Unstructured content, such as a product description or log message. You
+typically <<glossary-analysis,analyze>> text is for better search. See
```
Comment (lockewritesdocs) [inline docs/reference/glossary.asciidoc:500]:
```suggestion
typically <<glossary-analysis,analyze>> text for better search. See
```

**Review by lockewritesdocs — APPROVED** [review]
Left a few suggestions for typos, but looks great otherwise! 🎉

**Comment by jrodewig** [comment]
Thanks @lockewritesdocs. Much appreciated!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**